src/ball_tree.jl

```diff
@@ -15,87 +15,109 @@ end
 # When we create the bounding spheres we need some temporary arrays.
 # We create a type to hold them to not allocate these arrays at every
 # function call and to reduce the number of parameters in the tree builder.
-immutable ArrayBuffers{T <: AbstractFloat}
-    left::Vector{T}
-    right::Vector{T}
-    v12::Vector{T}
-    zerobuf::Vector{T}
+immutable ArrayBuffers{N, T <: AbstractFloat}
+    left::MVector{N, T}
```

Comment:
As a general comment, using `SVector` should be quite a bit faster than `MVector` (but often requires *significantly* more effort in changing the code). 

Are we pretty confident that `N` is small?